### PR TITLE
TradeOffer documentation

### PR DESCRIPTION
**Pull request by ST-DDT**
# TradeOffer System

Creates some documentation on the trade offer system. Fixes: https://github.com/SpongePowered/SpongeDocs/issues/434

If you would like to help me just go ahead.

## Links
* [SpongeAPI-Merchant](https://github.com/SpongePowered/SpongeAPI/tree/bleeding/src/main/java/org/spongepowered/api/item/merchant)

### CODE REVIEW

File: source/plugin/trade-offers.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+VillagerRegistry
+================
+
+The ``VillagerRegistry`` can be obtained from the ``GameRegistry``. It will be used to configure the
+``TradeOfferListMutator`` that will be applied on a ``Villager``'s level-up .
+
+.. code-block:: java
+
+    GameRegistry registry = this.game.getRegistry();
```
Comment (kashike) [inline source/plugin/trade-offers.rst]: Is it necessary to have this variable? Simply doing `VillagerRegistry villagerRegistry = this.game.getRegistry().getVillagerRegistry();` should be fine.

Comment (ST-DDT) [inline source/plugin/trade-offers.rst]: Fixed.

**Comment by Inscrutable** [comment]
Just FYI @ST-DDT you have been invited to join the SpongeDocs arm of the SpongePowered organisation on GitHub! There should be an email somewhere you need to click on... and then you'll get access to the Spongie preview feature.

**Comment by ST-DDT** [comment]
If nobody objects I would like to merge this this weekend.

**Comment by 12AwsomeMan34** [comment]
@ST-DDT seems sphinx just needed an update. You can rebase this PR and the build should succeed.

**Pull request merged**